tests/tests_app/runners/test_cloud.py

```diff
@@ -1368,3 +1377,69 @@ def run(self):
 
     with pytest.raises(ValueError, match="You requested a custom base image for the Work with name"):
         _validate_build_spec_and_compute(Work())
+
+
+def test_programmatic_lightningignore(monkeypatch, caplog, tmpdir):
+    mock_client = mock.MagicMock()
+    mock_client.projects_service_list_memberships.return_value = V1ListMembershipsResponse(
+        memberships=[V1Membership(name="test-project", project_id="test-project-id")]
+    )
+    mock_client.lightningapp_instance_service_list_lightningapp_instances.return_value = (
+        V1ListLightningappInstancesResponse(lightningapps=[])
+    )
+    mock_client.lightningapp_v2_service_create_lightningapp_release.return_value = V1LightningappRelease(
+        cluster_id="test"
+    )
+    cloud_backend = mock.MagicMock(client=mock_client)
+    monkeypatch.setattr(backends, "CloudBackend", mock.MagicMock(return_value=cloud_backend))
+
+    class MyWork(LightningWork):
+        def __init__(self):
+            super().__init__()
+            self.lightningignore = ["foo", "lightning_logs"]
+
+        def run(self):
+            # this is ignored
+            self.lightningignore.append("foobar")
```

Comment:
This should raise an exception.

Reply:
How do you suggest that I do it? Is there a property like `self.app.already_uploaded` that I can access inside the setter?

Reply:
You can add the condition within the setattr. If this is modified at runtime, it should raise an exception.

Reply:
Done!